Add staff grid component

bbaker6225 opened the pull request:
Early WIP PR for the staff grid. Let me know if there are any early concerns, or if we have more details about what info we want to show in the regular view
<img width="1593" alt="Screen Shot 2019-03-11 at 11 16 33 AM" src="https://user-images.githubusercontent.com/14864970/54134898-2811a300-43ef-11e9-9eeb-a49c1c1c1e07.png">

bbaker6225 commented on src/components/StaffGrid/index.jsx:
```diff
@@ -0,0 +1,68 @@
+  </div>
+);
+
+StaffGrid.propTypes = {
```
The prop types here are definitely excessive, I was just trying to capture all the possible data from the API, in case we do need any of this information elsewhere. Additionally, I tried to capture what seemed like required fields based off discovery admin, but I might be wrong

mikix replied:
Yeah, I usually just leave it at PropTypes.shape(), but I know there are good arguments for being more explicit.

AlasdairSwan commented on src/components/StaffGrid/index.jsx:
```diff
@@ -0,0 +1,68 @@
+  <div className="staff-grid row">
+    {staff.map(staffer => (
+      <div className="staffer-wrapper col-6 col-md-4 col-lg-3" key={staffer.uuid}>
+        <img src={staffer.profile_image_url} className="rounded-circle w-50" alt="" />
```
I see you're putting the border-radius on the images themselves. While working on the updated program details pages I noticed that not all of the instructor images are in the preferred format (see the last instructor here https://sandbox-edx-prospectus.edx.org/micromasters/columbiax-business-analytics). So to prevent those ones looking broken I put the border-radius and width on the wrapper, and use the ```overflow-hidden``` class.

bbaker6225 replied:
Happy to make these changes, but on the sandbox I'm seeing the same thing as this, except with `overflow:hidden` in the css for the image wrapper

AlasdairSwan replied:
Yeah, for the experiment on the sandbox I'm actually overriding the classes on ```img``` instead of changing universally. I should update that.

bbaker6225 replied:
Still not completely sure what changes you want here. When I move the `rounded-circle w-50` utility classes up to the wrapper along with `overflow-hidden`, that causes some funkiness

AlasdairSwan commented on src/components/StaffGrid/index.jsx:
```diff
@@ -13,10 +13,12 @@ const StaffGrid = ({ staff }) => (
           <Icon
             id="delete-icon"
             className={['mr-1', 'fa', 'fa-trash', 'fa-fw', 'text-danger']}
+            screenReaderText="Remove staffer"
```
It would be more meaningful to include the staffer's name.

AlasdairSwan commented on src/components/StaffGrid/index.jsx:
```diff
@@ -13,10 +13,12 @@ const StaffGrid = ({ staff }) => (
           <Icon
             id="edit-icon"
             className={['mr-1', 'fa', 'fa-edit', 'fa-fw']}
+            screenReaderText="Edit staffer"
```
It would be more meaningful to include the staffer's name.

bbaker6225 commented on src/components/StaffGrid/StaffGrid.test.jsx:
```diff
@@ -0,0 +1,39 @@
+];
+
+describe('CourseTable', () => {
+  it('renders each member of the staff', () => {
```
This test feels more useful to me than a snapshot, but I realize I am going against the grain of the rest of publisher's tests so I'm happy to change it

bbaker6225 commented:
@mikix @AlasdairSwan I changed the test to a snapshot to be consistent with the rest of publisher, and removed the grid from actually being used anywhere. Should be ready for a real review so @dianakhuang and I can more easily work off this for re-ordering and deleting
